pkg/compact/compact.go

```diff
@@ -764,7 +766,10 @@ func (cg *Group) Compact(ctx context.Context, dir string, planner Planner, comp
 		return false, ulid.ULID{}, errors.Wrap(err, "create compaction group dir")
 	}
 
-	shouldRerun, compID, err := cg.compact(ctx, subDir, planner, comp)
+	var err error
+	tracing.DoInSpan(ctx, "group_compaction", func(ctx context.Context) {
+		shouldRerun, compID, err = cg.compact(ctx, subDir, planner, comp)
+	}, opentracing.Tags{"group key": cg.Key()})
```

Comment:
Update tag name

Reply:
I thought `Key()` returned the key of the current group and hence named it accordingly. What's a more suitable name? Sorry I'm not sure.

Reply:
I mean the tag name should be group.key instead of group key

Reply:
I've done this. I think it's ready for an upstream PR now.

Reply:
Still some comments to address. After those are done then you can open a pr.